src/app/filters/filter-fields.component.html

```diff
@@ -14,16 +14,17 @@
         </li>
       </ul>
     </div>
-    <div *ngIf="currentField.type !== 'select'">
+    <div *ngIf="!currentField?.type || currentField.type === 'text' || currentField === 'default'">
       <input class="form-control" type="{{currentField.type}}" [(ngModel)]="currentValue"
              placeholder="{{currentField.placeholder}}"
              (keypress)="onValueKeyPress($event)"/>
     </div>
     <div *ngIf="currentField.type === 'select'">
       <div class="btn-group bootstrap-select form-control filter-select" dropdown>
-        <button type="button" class="btn btn-default dropdown-toggle" dropdownToggle>
+        <button type="button"
+          class="btn btn-default dropdown-toggle"
+          (click)="emitSelectedField()" dropdownToggle>
           <span class="filter-option pull-left">{{currentValue || currentField.placeholder}}</span>
-          <span class="caret"></span>
```

Comment:
I tried the example, but the "Filter by birth month" example no longer works. We cannot simply remove the caret in favor of type-ahead functionality. The existing functionality is still used elsewhere and we cannot break that.

The hot key functionality is also broken. I should be able to type 'j' and jump to the "January" option. Type "j" again and I should be able to jump to "July", "June", and so on.